pkg/frontend/querymiddleware/limits.go

```diff
@@ -124,6 +128,22 @@ func (l limitsMiddleware) Do(ctx context.Context, r Request) (Response, error) {
 		}
 	}
 
+	// Enforce the max end time.
+	creationGracePeriod := validation.SmallestPositiveNonZeroDurationPerTenant(tenantIDs, l.CreationGracePeriod)
```

Comment:
I think this should be `LargestPositiveNonZeroDurationPerTenant` instead, as if there's a tenant that has a higher grace period, we should be able to look into that.

Reply:
Fixed. But I have a question maybe it is stupid: Why don't we use the corresponding setting for each tenant?

Reply:
Since we don't decide here which blocks have to be queried, we can't be that granular, so we apply the least conservative truncation. Does that make sense?

Reply:
Make sense. Thanks!